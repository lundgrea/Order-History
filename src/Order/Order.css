img{
  height: 50px;
  width: 50px;
  float: left
}

.card {
  display: flex;
  border-radius: 10px;
  margin: 20px;
  width: 800px;
  background: white;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

button{
  background: red;
  color: white;
}