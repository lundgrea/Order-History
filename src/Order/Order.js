import React from 'react';
import './Order.css';


const Order = ({name, price, description, id, img}) => {
  console.log(name)
  return (
    <section>
      <img alt="item-photo" src={img}></img>
      <h2>{name}</h2>
      <h2>{price}</h2>
      <p>{description}</p>
      <button>Remove From History</button>
    </section>
  )
}

export default Order