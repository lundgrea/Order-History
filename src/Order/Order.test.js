import React from 'react';
import { shallow } from 'enzyme';
import Order from './Order';

describe('Order', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<Order
      img="https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154"
      name="iphone"
      price="300"
      description="One of a kind iphone"
      removeOrder={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call the removeOrder prop with the card\'s id when clicked', () =>{
    const removeOrderMock = jest.fn()
    const wrapper = shallow(<Order
      img="https://jangal.com/api/ui/image/watermark?path=/uploads/product/Great%20Expectations%20by%20Charles%20Dickens.png&etype=product"
      name="A Book"
      price="15"
      description="very interesting book"
      id={7}
      removeOrder={removeOrderMock}
    />
    )
    wrapper.find('.item_remove-button').simulate('click')

    expect(removeOrderMock).toHaveBeenCalledWith(7)
  })
})