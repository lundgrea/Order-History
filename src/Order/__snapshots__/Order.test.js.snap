// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match the snapshot with all data passed in correctly 1`] = `
<section
  className="card"
>
  <img
    alt="item"
    className="item_image"
    src="https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/iphone-xs-max-gold-select-2018?wid=940&hei=1112&fmt=png-alpha&qlt=80&.v=1550795409154"
  />
  <h2
    className="item_name"
  >
    iphone
  </h2>
  <h3
    className="item_price"
  >
    300
  </h3>
  <p
    className="item_description"
  >
    One of a kind iphone
  </p>
  <button
    className="item_remove-button"
    onClick={[Function]}
  >
    Remove From History
  </button>
</section>
`;
